components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -663,4 +671,100 @@ private static boolean revokeLatestTokensWithScopes(Set<String> scopes, String c
         }
         return true;
     }
+
+    /**
+     * Resolve user.
+     *
+     * @param tenantDomain The tenant domain which user is trying to access.
+     * @param username     The username of resolving user.
+     * @return User object.
+     * @throws IdentityApplicationManagementException Error when user cannot be resolved.
+     */
+    public static Optional<User> getUser(String tenantDomain, String username)
+            throws IdentityApplicationManagementException {
+
+        User user = null;
+        try {
+            int tenantID = IdentityTenantUtil.getTenantId(tenantDomain);
+            String userId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserId();
+
+            if (tenantID == MultitenantConstants.SUPER_TENANT_ID) {
+                user = getUserFromTenant(username, userId, tenantID);
+            } else {
+                Tenant tenant = OAuthComponentServiceHolder.getInstance().getRealmService()
+                        .getTenantManager().getTenant(tenantID);
+                String accessedOrganizationId = tenant.getAssociatedOrganizationUUID();
+                if (accessedOrganizationId == null) {
```

Comment:
Shall we use `StringUtils.isEmpty(String str)`, since it will inspect for both empty and null check

Reply:
resolved by 57676650f4bb2e0fb744eae58ff5aebe4aa65b2f